Open in finder

PavelLaptev opened the pull request:
## 🧢 Changes

There are two `FileContextMenu.svelte` files. I only updated the one for V3. We should probably refactor and remove the old one later, but I’ll do that in a separate PR.

Added "Open In Finder" in two places.

<img width="566" height="268" alt="image" src="https://github.com/user-attachments/assets/7e6415b9-92cf-4a9e-98ff-e1dcb0e552fd" />
<img width="280" height="181" alt="image" src="https://github.com/user-attachments/assets/87693e42-daa7-497a-9323-12f7fe215042" />


## ☕️ Reasoning

<!--
If this PR is related to a specific issue, uncomment this section
and link it via the following text:

## 🎫 Affected issues

Fixes: INSERT_ISSUE_NUMBER

-->

<!--
If this is a WIP PR and you have todos left, feel free to uncomment this and turn this PR into a draft, see https://github.blog/2019-02-14-introducing-draft-pull-requests/

## 📌 Todos

-->

PavelLaptev committed:
feat: add "Show in Finder" option to file context menus for easier file access

PavelLaptev committed:
feat(menu): add cross-platform "Show in Finder" project menu item and command

Caleb-T-Owens commented on apps/desktop/src/components/v3/FileContextMenu.svelte:
```diff
@@ -373,6 +386,25 @@
+						<ContextMenuItem
+							label={showInFolderLabel}
+							onclick={async () => {
+								try {
+									const project = await projectService.fetchProject(projectId);
+									const projectPath = project?.path;
+									if (projectPath) {
+										const absPath = await join(projectPath, changes[0]!.path);
+										await showFileInFolder(absPath);
+									}
+									contextMenu.close();
+								} catch (error) {
+									console.error('Failed to show in file manager:', error);
+									toasts.error('Failed to show in file manager');
+								}
```
If we removed this try/catch it would mean the error is handled by the global error handler, which means it would get fed into posthog so we could get some details about how many users this fails for.

Using a toast might be a design choice  though... though I wonder if we could do something that lets it still go through the global error handler so we still get the posthog reports

PavelLaptev committed:
Enhance cross-platform file manager functionality

Implemented a polished "Show in Finder/Explorer/File Manager" feature for GitButler, ensuring seamless cross-platform support and integrating feedback from code review. This change replaces dynamic imports with a centralized utility for improved organization and error handling.

- Utilized the centralized showFileInFolder utility in ProjectSettingsMenuAction.svelte to replace direct Tauri invoke calls, eliminating dynamic import usage.
- Enhanced error handling and added platform-specific UI labels for better user experience.
- Resolved Rust code issues by converting &Path to &str, and removed macOS-only restriction to broaden feature availability.
- Improved menu organization by adding a separator for clarity.

PavelLaptev committed:
Refactor error handling for file navigator

Caleb's code review highlighted the redundancy of having try/catch blocks within the showFileInFolder utility. The code has been refactored to leverage global error handling, which simplifies the code and ensures that analytics are captured effectively via posthog.

- Removed unnecessary try/catch in the showFileInFolder utility to rely on global error handling.
- Updated FileContextMenu to display an error toast, while allowing the global error handler to report issues for better analytics integration and user feedback.
- This change addresses Caleb's feedback and improves the robustness of error reporting.
